src/components/data-management/CellLevelUploadModal.jsx

```diff
@@ -218,52 +220,73 @@ const CellLevelUploadModal = (props) => {
       </Row>
       <Row>
         <Col span={24}>
-          {/* eslint-enable react/jsx-props-no-spreading */}
-
-          {filesList.length ? (
+          {file ? (
             <>
               <Divider orientation='center'>To upload</Divider>
               <List
-                dataSource={filesList}
                 size='small'
                 itemLayout='horizontal'
                 grid='{column: 4}'
-                renderItem={(file) => (
-
-                  <List.Item
-                    key={file.name}
-                    style={{ width: '100%' }}
-                  >
-                    <Space>
-
-                      <CheckCircleTwoTone twoToneColor='#52c41a' />
-
-                      <Text
-                        ellipsis={{ tooltip: file.name }}
-                        style={{ width: '200px' }}
-                      >
-                        {file.name}
-
-                      </Text>
-                      <DeleteOutlined style={{ color: 'crimson' }} onClick={() => { removeFile(file.name); }} />
-                    </Space>
-                  </List.Item>
-
-                )}
-              />
+              >
+                <List.Item
+                  key={file.name}
+                  style={{ width: '100%' }}
+                >
+                  <Space>
+                    <CheckCircleTwoTone twoToneColor='#52c41a' />
+                    <Text
+                      ellipsis={{ tooltip: file.name }}
+                      style={{ width: '200px' }}
+                    >
+                      {file.name}
+                    </Text>
+                    <DeleteOutlined style={{ color: 'crimson' }} onClick={() => { setFile(false); }} />
+                  </Space>
+                </List.Item>
+              </List>
             </>
           ) : ''}
         </Col>
       </Row>
     </Modal>
-
   );
-};
 
+  if (Object.keys(cellLevelMetadata).length) {
+    const {
+      name, size, uploadStatus, percentProgress, createdAt,
+    } = cellLevelMetadata;
+
+    const fileInfoObject = {
+      name,
+      upload: {
+        status: uploadStatus,
+        progress: percentProgress,
+      },
+      size,
+      fileObject: file.fileObject,
+      lastModified: createdAt,
+    };
+    return (
+      <UploadDetailsModal
+        onCancel={onCancel}
+        onUpload={onUploadFile}
+        onDownload={downloadData}
+        onDelete={() => dispatch(deleteCellLevelMetadata(activeExperimentId))}
+        onRetry={() => onUploadFile(file)}
```

Comment:
we don't need `retry` now, do we?

Reply:
We need it, the modal shows upload progress so retry should be there, just delete instead of replace